### ISSUE TITLE
Problem installing php-git on OSX 10.7.5

### ISSUE THREAD

**Issue by spemmelaar** [libgit2/php-git#55]
I have read a couple of other people's similar problems with this.
- I have successfully installed libgit2 (not using macports either).
- One thing which I am doing slightly differently in my installation is using the PHP 5.3 version which comes with MAMP.
  I am specifying this by typing: 
  ./configure --with-php-config=/Applications/MAMP/bin/php5.3/bin/php-config
- After this I am getting the following stack trace when it comes to running the 'make' command.

Shane-Pemmelaars-MacBook-Pro:php-git shanepemmelaar$ make
/bin/sh /Users/shanepemmelaar/Desktop/php-git/libtool --mode=compile cc  -I. -I/Users/shanepemmelaar/Desktop/php-git -DPHP_ATOM_INC -I/Users/shanepemmelaar/Desktop/php-git/include -I/Users/shanepemmelaar/Desktop/php-git/main -I/Users/shanepemmelaar/Desktop/php-git -I/Applications/MAMP/bin/php5.3/include/php -I/Applications/MAMP/bin/php5.3/include/php/main -I/Applications/MAMP/bin/php5.3/include/php/TSRM -I/Applications/MAMP/bin/php5.3/include/php/Zend -I/Applications/MAMP/bin/php5.3/include/php/ext -I/Applications/MAMP/bin/php5.3/include/php/ext/date/lib -I/Users/shanepemmelaar/Desktop/php-git/libgit2/include  -DHAVE_CONFIG_H  -g -O2 -Wunused-variable -Wpointer-sign -Wimplicit-function-declaration -Winline -Wunused-macros -Wredundant-decls -Wstrict-aliasing=2 -Wswitch-enum -Wdeclaration-after-statement   -c /Users/shanepemmelaar/Desktop/php-git/git2.c -o git2.lo 
mkdir .libs
 cc -I. -I/Users/shanepemmelaar/Desktop/php-git -DPHP_ATOM_INC -I/Users/shanepemmelaar/Desktop/php-git/include -I/Users/shanepemmelaar/Desktop/php-git/main -I/Users/shanepemmelaar/Desktop/php-git -I/Applications/MAMP/bin/php5.3/include/php -I/Applications/MAMP/bin/php5.3/include/php/main -I/Applications/MAMP/bin/php5.3/include/php/TSRM -I/Applications/MAMP/bin/php5.3/include/php/Zend -I/Applications/MAMP/bin/php5.3/include/php/ext -I/Applications/MAMP/bin/php5.3/include/php/ext/date/lib -I/Users/shanepemmelaar/Desktop/php-git/libgit2/include -DHAVE_CONFIG_H -g -O2 -Wunused-variable -Wpointer-sign -Wimplicit-function-declaration -Winline -Wunused-macros -Wredundant-decls -Wstrict-aliasing=2 -Wswitch-enum -Wdeclaration-after-statement -c /Users/shanepemmelaar/Desktop/php-git/git2.c  -fno-common -DPIC -o .libs/git2.o
In file included from /Users/shanepemmelaar/Desktop/php-git/git2.c:25:
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:35:19: error: php.h: No such file or directory
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:36:36: error: Zend/zend_interfaces.h: No such file or directory
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:37:36: error: Zend/zend_exceptions.h: No such file or directory
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:38:38: error: ext/spl/spl_exceptions.h: No such file or directory
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:39:39: error: ext/standard/php_string.h: No such file or directory
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:40:29: error: date/php_date.h: No such file or directory
In file included from /Users/shanepemmelaar/Desktop/php-git/git2.c:25:
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:45: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘git2_module_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:50: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:51: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:52: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:53: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:54: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:55: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:56: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:57: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:58: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:59: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:60: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:61: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:62: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:63: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:64: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:65: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:66: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘zend_class_entry’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:69: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:74: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:79: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:84: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:91: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:96: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:101: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:106: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:117: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:126: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:132: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:137: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:142: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:147: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:152: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:157: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:162: error: expected specifier-qualifier-list before ‘zend_object’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:168: error: expected specifier-qualifier-list before ‘zval’
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:204: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:205: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:206: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:207: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:208: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:210: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:212: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:225: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:267: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:287: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/php_git2.h:319: error: expected ‘;’, ‘,’ or ‘)’ before ‘TSRMLS_DC’
/Users/shanepemmelaar/Desktop/php-git/git2.c:26:31: error: ext/standard/info.h: No such file or directory
/Users/shanepemmelaar/Desktop/php-git/git2.c:28: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:29: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:30: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:31: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:32: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:33: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:34: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:35: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:36: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:37: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:38: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:39: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:40: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:41: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:42: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:43: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:44: warning: parameter names (without types) in function declaration
/Users/shanepemmelaar/Desktop/php-git/git2.c:46: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/git2.c:88: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/git2.c:170: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/git2.c:185: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/git2.c:200: error: expected ‘)’ before ‘_’ token
/Users/shanepemmelaar/Desktop/php-git/git2.c: In function ‘PHP_MINIT_FUNCTION’:
/Users/shanepemmelaar/Desktop/php-git/git2.c:219: error: ‘TSRMLS_C’ undeclared (first use in this function)
/Users/shanepemmelaar/Desktop/php-git/git2.c:219: error: (Each undeclared identifier is reported only once
/Users/shanepemmelaar/Desktop/php-git/git2.c:219: error: for each function it appears in.)
/Users/shanepemmelaar/Desktop/php-git/git2.c:237: error: ‘SUCCESS’ undeclared (first use in this function)
/Users/shanepemmelaar/Desktop/php-git/git2.c: In function ‘PHP_MINFO_FUNCTION’:
/Users/shanepemmelaar/Desktop/php-git/git2.c:242: warning: implicit declaration of function ‘php_printf’
/Users/shanepemmelaar/Desktop/php-git/git2.c:243: warning: implicit declaration of function ‘php_info_print_table_start’
/Users/shanepemmelaar/Desktop/php-git/git2.c:244: warning: implicit declaration of function ‘php_info_print_table_row’
/Users/shanepemmelaar/Desktop/php-git/git2.c:246: warning: implicit declaration of function ‘php_info_print_table_end’
/Users/shanepemmelaar/Desktop/php-git/git2.c: In function ‘PHP_RINIT_FUNCTION’:
/Users/shanepemmelaar/Desktop/php-git/git2.c:253: error: ‘SUCCESS’ undeclared (first use in this function)
/Users/shanepemmelaar/Desktop/php-git/git2.c: In function ‘PHP_RSHUTDOWN_FUNCTION’:
/Users/shanepemmelaar/Desktop/php-git/git2.c:259: error: ‘SUCCESS’ undeclared (first use in this function)
/Users/shanepemmelaar/Desktop/php-git/git2.c: At top level:
/Users/shanepemmelaar/Desktop/php-git/git2.c:262: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘**attribute**’ before ‘git2_module_entry’
/Users/shanepemmelaar/Desktop/php-git/git2.c: In function ‘ZEND_GET_MODULE’:
/Users/shanepemmelaar/Desktop/php-git/git2.c:280: error: expected ‘{’ at end of input
make: *_\* [git2.lo] Error 1
- Any help with this problem would be greatly appreciated so that I may integrate the use of git in with the php project I am working on.

Cheers,
Shane

**Comment by chobie** [comment]
it seems you don't have php includes files. have you installed XAMPP Developer Package?
I'm not sure about XAMPP installation so I recommend to try to build under phpenv & php-build env.

**Comment by spemmelaar** [comment]
Hi Chobie,
I was using MAMP server, not XAMPP. MAMP is at http://www.mamp.info/en/index.html

I resolved this problem by downloading XAMPP and setting up php-git  with the ./config --with-php-config=/Applications/XAMPP/xamppfiles/bin/php-config (XAMPP's php-config) and I managed to run the 'make' command without a compilation error. Hence the problem is resolved for me, but may exist if someone is trying to set this up using MAMP.

./configure --with-php-config=/Applications/XAMPP/xamppfiles/bin/php-config
